tools/IndexStore/CMakeLists.txt

```diff
@@ -49,6 +49,10 @@ if(APPLE)
 
   set_property(TARGET IndexStore APPEND_STRING PROPERTY
                LINK_FLAGS ${INDEXSTORE_LINK_FLAGS})
+else()
+  # Blocks are off by default on non-darwin platforms.
+  target_compile_options(IndexStore PRIVATE -fblocks)
```

Comment:
This can be hoisted out of the condition as it is safe to apply `-fblocks` when building for Darwin.  However, please ensure that you do not apply this when the C++ compiler is not clang.  The Windows builds use cl to build the toolchain as it is much faster and better.